packages/rules/src/mustSatisfyPeerDependencies.ts

```diff
@@ -0,0 +1,272 @@
+// does not accept `<` or `>`
+// TODO: accept minor pins `~4.2.1`
+export const RANGE_REGEX = /^(\*|x|\^?\d+(\.x|\.x\.x|\.\d+|\.\d+\.x|\.\d+\.\d+)?( \|\| \^?\d+(\.x|\.x\.x|\.\d+|\.\d+\.x|\.\d+\.\d+)?)*)$/;
+
+function checkSatisfyPeerDependencies(context: Context, opts: Options) {
+  const { skipUnparseableRanges } = opts;
+  const graphService = new PackageDependencyGraphService();
+  const rootNode = graphService.buildDependencyGraph(path.resolve(context.getPackageJsonPath()), 1);
+  const rootDependencies = rootNode.packageJson.dependencies || {};
+  const rootPeerDependencies = rootNode.packageJson.peerDependencies || {};
+  const rootPackageJsonPath = rootNode.paths.packageJsonPath;
+  const rootPackageName = rootNode.packageJson.name || rootPackageJsonPath;
+
+  for (const [peerDependencyName, peerDependencyRange] of Object.entries(rootPeerDependencies)) {
+    const dependencyRange = rootDependencies[peerDependencyName];
+    if (dependencyRange != null) {
+      context.addError({
+        file: rootPackageJsonPath,
+        message: `[0] Package ${rootPackageName} has overloaded ${peerDependencyName} dependencies.`,
+        longMessage: `Peer dependency '${peerDependencyRange}' and regular dependency '${dependencyRange}'.`,
+      });
+    }
+  }
+
+  interface IPeerDependencyRequirement {
+    node: IPackageDependencyGraphNode;
+    range: string;
+  }
+  const allRequiredPeerDependencies: { [peerDependencyName: string]: IPeerDependencyRequirement[] } = {};
+
+  for (const [, dependencyNode] of rootNode.dependencies) {
+    const requiredPeerDependencies = dependencyNode.packageJson.peerDependencies;
+    if (requiredPeerDependencies == null) {
+      continue;
+    }
+    for (const [peerDependencyName, range] of Object.entries(requiredPeerDependencies)) {
+      if (!RANGE_REGEX.test(range)) {
+        if (skipUnparseableRanges) {
+          continue;
+        }
+        throw new Error(
+          `Unable to parse dependency range ${peerDependencyName} @ ${range} in ${dependencyNode.paths.packageJsonPath}`
+        );
+      }
+      if (allRequiredPeerDependencies[peerDependencyName] == null) {
+        allRequiredPeerDependencies[peerDependencyName] = [];
+      }
+      allRequiredPeerDependencies[peerDependencyName].push({ node: dependencyNode, range });
+    }
+  }
+
+  for (const [peerDependencyName, peerDependencyRequirements] of Object.entries(allRequiredPeerDependencies)) {
+    let mostStrictPeerRequirement: IPeerDependencyRequirement = peerDependencyRequirements[0];
+    for (const peerRequirement of peerDependencyRequirements) {
+      if (peerRequirement === mostStrictPeerRequirement) {
+        continue;
+      } else {
+        if (doesASatisfyB(peerRequirement.range, mostStrictPeerRequirement.range)) {
+          mostStrictPeerRequirement = peerRequirement;
+        } else if (doesASatisfyB(mostStrictPeerRequirement.range, peerRequirement.range)) {
```

Comment:
i feel like you could just do `else if (!doesASatisfyB(mostStrictPeerRequirement.range, peerRequirement.range)) {` and skip this `continue`?